tools/pvacvector/run.py

```diff
@@ -278,23 +278,33 @@ def find_optimal_path(Paths, distance_matrix, seq_dict, seq_keys, base_output_di
         f.write(''.join(output))
     return (results_file, "")
 
-def shorten_problematic_peptides(input_file, problematic_start, problematic_end, output_dir):
-    print("Shortening problematic peptides")
+def shorten_problematic_peptides(input_file, problematic_start, problematic_end, output_dir, min_peptide_length):
     records = []
+    shortened = False
     for record in SeqIO.parse(input_file, "fasta"):
-        if record.id in problematic_start and record.id in problematic_end:
-            record_new = SeqRecord(Seq(str(record.seq)[1:-1], IUPAC.protein), id=record.id, description=record.description)
-        elif record.id in problematic_start:
-            record_new = SeqRecord(Seq(str(record.seq)[1:], IUPAC.protein), id=record.id, description=record.description)
-        elif record.id in problematic_end:
-            record_new = SeqRecord(Seq(str(record.seq)[:-1], IUPAC.protein), id=record.id, description=record.description)
+        if record.id not in problematic_start and record.id not in problematic_end:
+            records.append(record)
         else:
-            record_new = record
-        records.append(record_new)
-    os.makedirs(output_dir, exist_ok=True)
-    new_input_file = os.path.join(output_dir, "vector_input.fa")
-    SeqIO.write(records, new_input_file, "fasta")
-    return new_input_file
+            if record.id in problematic_start and record.id in problematic_end:
```

Comment:
Let's say that we have a peptide that's 33aa and a max-length of 32aa.  If both the start and end are labelled problematic, we will never do any trimming of either end (because it will try to trim both at once, and drop below the min length).  It would be nice to alter this such that it trims the front first, then checks length, then trims the end, then checks length again (so at least one end would get trimmed in the scenario I laid out).